crates/solidity/outputs/cargo/crate/src/generated/bindings/generated/built_ins/0.8.26.sol

```diff
@@ -0,0 +1,106 @@
+// This file is generated automatically by infrastructure scripts. Please don't edit by hand.
+
+contract $BuiltIns$ {
+  function addmod(uint x, uint y, uint k) public returns (uint);
+  function assert(bool condition) public;
+  function blockhash(uint blockNumber) public returns (bytes32);
+  function blobhash(uint index) public returns (bytes32);
+  function ecrecover(bytes32 hash, uint8 v, bytes32 r, bytes32 s) public returns (address);
+  function gasleft() public returns (uint256);
+  function keccak256(bytes memory) public returns (bytes32);
+  function mulmod(uint x, uint y, uint k) public returns (uint);
+  function require(bool condition) public;
+  function require(bool condition, string memory message) public;
+  function require(bool condition, Error error) public;
+  function revert() public;
+  function revert(string memory reason) public;
+  function ripemd160(bytes memory) public returns (bytes20);
+  function selfdestruct(address payable recipient) public;
+  function sha256(bytes memory) public returns (bytes32);
+  struct $abiType {
+    function(bytes memory, $args) returns ($args) decode;
+    function($args) returns (bytes memory) encode;
+    function(function(), $args) returns (bytes memory) encodeCall;
+    function($args) returns (bytes memory) encodePacked;
+    function(bytes4 selector, $args) returns (bytes memory) encodeWithSelector;
+    function(string memory, $args) returns (bytes memory) encodeWithSignature;
+  }
+  struct $address {
+    uint256 balance;
+    bytes code;
+    bytes32 codehash;
+    function(bytes memory) returns (bool, bytes memory) call;
+    function(bytes memory) returns (bool, bytes memory) delegatecall;
+    function(uint256) returns (bool) send;
+    function(bytes memory) returns (bool, bytes memory) staticcall;
+    function(uint256) transfer;
+  }
+  struct $array {
+    uint length;
+    function(uint) returns ($element) $index;
+    function() returns ($element) push;
+    function($element) push;
+    function() pop;
+  }
+  struct $arrayFixed {
+    uint length;
+    function(uint) returns ($element) $index;
+  }
+  struct $blockType {
+    uint basefee;
+    uint blobbasefee;
+    uint chainid;
+    address payable coinbase;
+    uint difficulty;
+    uint gaslimit;
+    uint number;
+    uint prevrandao;
+    uint timestamp;
+  }
+  struct $bytes {
+    function($args) returns (bytes memory) concat;
+  }
+  struct $callOptions {
+    uint gas;
+    uint salt;
+    uint value;
+  }
+  struct $functionExternal {
```

Comment:
WDYT of using PascalCase and full type names? is the shorthand used here defined/used somewhere in Solidity itself?

- `msgType` -> `MessageType`
- `txType` -> `TransactionType`
- etc...

Reply:
The names are completely made up. I was trying to have some level of consistency in mapping the origin of the names, but I probably introduced some conflicts along the way. For `msgType` and `txType` for example, those are the types of the `msg` and `tx` global variables, hence the short camel case form. For functions, arrays and other built-in types, it's convenient because we need to generate them from the rules, and we usually would generate the type by prefixing the type name with `%` (eg. `%address`, or `%uint256`). Some variants such as `%functionExternal` and `%arrayFixed` are derivatives of this rule.

Reply:
I wonder if that can be solved by the additional built-ins rules file? as we don't need to have this prefix?
I'll keep thinking about this. Not blocking to this PR of course.